airflow-core/src/airflow/logging/remote.py

```diff
@@ -44,6 +44,15 @@ def upload(self, path: os.PathLike | str, ti: RuntimeTI) -> None:
         """Upload the given log path to the remote storage."""
         ...
 
-    def read(self, relative_path: str, ti: RuntimeTI) -> tuple[LogSourceInfo, LogMessages | None]:
+    def read(self, relative_path: str, ti: RuntimeTI) -> LegacyLogResponse:
         """Read logs from the given remote log path."""
         ...
+
+
+@runtime_checkable
+class RemoteLogStreamIO(Protocol):
```

Comment:
I think this protocol should be a extension to the non-streaming interface, so:

```suggestion
class RemoteLogStreamIO(RemoteLogIO):
```

Reply:
I had tried using subclass of `RemoteLogIO` before, but it raise some Mypy error.
If then, do we still want to make it subclass here and add the ignore mypy comment?

Reply:
Either option works for me, but I’ll wait for your suggestion, Ash.

Reply:
Sorry missed this message.

I think it depends on what the Mypy error message was?

Maybe it needs to be `RemoteLogStreamIO(RemoteLogIO, Protocol):` to keep mypy happy?

Reply:
Thanks for the hint! It works well with inheriting both `RemoteLogIO` and `Protocol`.